python_ta/contracts/__init__.py

```diff
@@ -603,3 +583,39 @@ def _debug(msg: str) -> None:
         return
 
     print("[PyTA]", msg, file=sys.stderr)
+
+
+def _set_invariants(klass: type) -> None:
+    """Retrieve and set the representation invariants of this class"""
+    # Update representation invariants from this class' docstring and those of its superclasses.
+    rep_invariants: List[Tuple[str, CodeType]] = []
+
+    # Iterate over all inherited classes except builtins
+    for cls in reversed(klass.__mro__):
+        if "__representation_invariants__" in cls.__dict__:
+            rep_invariants.extend(cls.__representation_invariants__)
+        elif cls.__module__ != "builtins":
+            assertions = parse_assertions(cls, parse_token="Representation Invariant")
+            # Try compiling assertions
+            for assertion in assertions:
+                try:
+                    compiled = compile(assertion, "<string>", "eval")
+                except:
+                    _debug(
+                        f"Warning: representation invariant {assertion} could not be parsed as a valid Python expression"
+                    )
+                    continue
+                rep_invariants.append((assertion, compiled))
+
+    setattr(klass, "__representation_invariants__", rep_invariants)
+
+
+def check_invariants(obj: object) -> None:
```

Comment:
Okay, so reviewing the documentation made me realize that `check_...` is actually a bad name for this function, because it's too similar to `check_contracts`, and they do fundamentally different things. I actually slightly regret choosing `check_contracts` as the name of the decorator, since that function is *enabling*, but not *checking*, RIs. But we're stuck with that.

Instead please rename this function to `validate`, which is another common verb used for this type of action.